src/Native/Unix/gen-buildsys-gcc.sh

```diff
@@ -0,0 +1,146 @@
+    exit 1
+fi
+
+if [ -z "$CC" ]; then
+  CC="$(command -v "${gcc_prefix}"gcc"$desired_gcc_version")"
+  export CC
+fi
+
+build_arch="$4"
+buildtype=DEBUG
+generator="Unix Makefiles"
+__UnprocessedCMakeArgs=""
+
+ITER=-1
+for i in "$@"; do
+    ITER=$((ITER + 1))
+    if [ $ITER -lt 5 ]; then continue; fi
+    upperI="$(echo "$i" | awk '{print toupper($0)}')"
+    case $upperI in
+      # Possible build types are DEBUG, CHECKED, RELEASE, RELWITHDEBINFO, MINSIZEREL.
+      DEBUG | CHECKED | RELEASE | RELWITHDEBINFO | MINSIZEREL)
+      buildtype=$upperI
+      ;;
+      NINJA)
+      generator=Ninja
+      ;;
+      *)
+      __UnprocessedCMakeArgs="${__UnprocessedCMakeArgs}${__UnprocessedCMakeArgs:+ }$i"
+    esac
+done
+
+OS=$(uname)
+
+locate_gcc_exec() {
+  if command -v "$gcc_prefix$1$desired_gcc_version" > /dev/null 2>&1
+  then
+    command -v "$gcc_prefix$1$desired_gcc_version"
+  elif command -v "$gcc_prefix$1" > /dev/null 2>&1
+  then
+    command -v "$gcc_prefix$1"
+  else
+    exit 1
+  fi
+}
+
+if ! gcc_ar="$(locate_gcc_exec ar)"; then { echo "Unable to locate gcc-ar"; exit 1; } fi
+
+if [ -z "$NM" ]; then
+  if ! gcc_nm="$(locate_gcc_exec nm)"; then { echo "Unable to locate gcc-nm"; exit 1; } fi
+else
+  gcc_nm="$NM"
+fi
+
+if [ "$OS" = "Linux" ] || [ "$OS" = "FreeBSD" ] || [ "$OS" = "OpenBSD" ] || [ "$OS" = "NetBSD" ] || [ "$OS" = "SunOS" ]; then
+  if [ -z "$OBJDUMP" ]; then
+    if ! gcc_objdump="$(locate_gcc_exec objdump)"; then { echo "Unable to locate gcc-objdump"; exit 1; } fi
+  else
+    gcc_objdump="$OBJDUMP"
+  fi
+fi
+
+if [ -z "$OBJCOPY" ]; then
+  if ! gcc_objcopy="$(locate_gcc_exec objcopy)"; then { echo "Unable to locate gcc-objcopy"; exit 1; } fi
+else
+  gcc_objcopy="$OBJCOPY"
+fi
+
+if ! gcc_ranlib="$(locate_gcc_exec ranlib)"; then { echo "Unable to locate gcc-ranlib"; exit 1; } fi
+
+cmake_extra_defines=
+if [ -n "$LLDB_LIB_DIR" ]; then
+    cmake_extra_defines="$cmake_extra_defines -DWITH_LLDB_LIBS=$LLDB_LIB_DIR"
+fi
+if [ -n "$LLDB_INCLUDE_DIR" ]; then
+    cmake_extra_defines="$cmake_extra_defines -DWITH_LLDB_INCLUDES=$LLDB_INCLUDE_DIR"
+fi
+if [ "$CROSSCOMPILE" = "1" ]; then
+    if [ -z "$ROOTFS_DIR" ]; then
+        echo "ROOTFS_DIR not set for crosscompile"
+        exit 1
+    fi
+    if [ -z "$CONFIG_DIR" ]; then
+        CONFIG_DIR="$1/cross"
+    fi
+    export TARGET_BUILD_ARCH=$build_arch
+    cmake_extra_defines="$cmake_extra_defines -C $CONFIG_DIR/tryrun.cmake"
+    cmake_extra_defines="$cmake_extra_defines -DCMAKE_TOOLCHAIN_FILE=$CONFIG_DIR/toolchain.cmake"
+fi
+if [ "$build_arch" = "armel" ]; then
+    cmake_extra_defines="$cmake_extra_defines -DARM_SOFTFP=1"
+fi
+
+cmake \
+  -G "$generator" \
+  "-DCMAKE_AR=$gcc_ar" \
+  "-DCMAKE_NM=$gcc_nm" \
+  "-DCMAKE_RANLIB=$gcc_ranlib" \
+  "-DCMAKE_OBJCOPY=$gcc_objcopy" \
+  "-DCMAKE_OBJDUMP=$gcc_objdump" \
+  "-DCMAKE_BUILD_TYPE=$buildtype" \
+  "-DCMAKE_EXPORT_COMPILE_COMMANDS=1 " \
+  "$cmake_extra_defines" \
```

Comment:
remove the comments here too

Reply:
I have synchronized script with CoreCLR, modulo the unused options like coverage, generator etc.